Sources/StreamChatUI/ChatChannel/ChatChannelVC.swift

```diff
@@ -117,6 +117,19 @@ open class ChatChannelVC: _ViewController,
 
         setChannelControllerToComposerIfNeeded(cid: channelController.cid)
 
+        // If the given message id is a reply that is inside a thread, we need
+        // to fetch the parent message, jump to the parent message and then open
+        // the thread so that we can jump to the thread reply.
+        // For this, we need to manipulate the original channel controller to contain
+        // the parent message id instead of the reply id.
+        var initialReplyId: MessageId?
+        if let initialMessageId = channelController.channelQuery.pagination?.parameter?.aroundMessageId,
+           let message = channelController.dataStore.message(id: initialMessageId),
```

Comment:
Yes, I thought about this as well, but it is very unlikely the message won't be on the database since when coming from a Push Notification, we load the message into the database.

I didn't want to add this built-in to avoid complexity. But we can add this later if we think customers need it. Although they can do this themselves, they can check if the message is available locally with the `dataStore` property, and if it isn't, they can load it with `messageController`.

I have a feeling that right now, it is not worth adding this fallback builtin, especially since customers can easily overcome this, but let me know what you think 🙂

Reply:
Yeah, makes sense!